tracer/src/Datadog.Trace/AppSec/Rcm/AsmDataProduct.cs

```diff
@@ -0,0 +1,43 @@
+﻿// <copyright file="AsmDataProduct.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+using System.Collections.Generic;
+using Datadog.Trace.AppSec.Rcm.Models.AsmData;
+using Datadog.Trace.RemoteConfigurationManagement;
+
+namespace Datadog.Trace.AppSec.Rcm;
+
+internal class AsmDataProduct : IAsmConfigUpdater
+{
+    public void ProcessUpdates(ConfigurationStatus configurationStatus, List<RemoteConfiguration> files)
```

Comment:
I don't like the idea of passing the configuration status in input and modify it. I believe we would need a configupdater that would update the config passing Product config. Something like a builder class.

```
var configUpdater = new ConfigUpdater();
configUpdater.WithAsmDataProduct(files, filesToRemove);
configUpdater.WithAsmProduct(files, filesToRemove);
configUpdater.WithAsmFeaturesProduct(files, filesToRemove);
```

It was a bit too much to change in the same PR. So maybe we can go with this for now and refactor in a different PR (if you agree with the idea obv)